table/tables/tables.go

```diff
@@ -925,7 +925,8 @@ func (t *tableCommon) AllocAutoID(ctx sessionctx.Context) (int64, error) {
 		return 0, err
 	}
 	if t.meta.ShardRowIDBits > 0 {
-		if t.overflowShardBits(rowID) {
+		// use max record ShardRowIDBits to check  overflow.
+		if OverflowShardBits(rowID, t.meta.MaxShardRowIDBits) {
```

Comment:
Could we use `ShardRowIDBits` instead of `MaxShardRowIDBits`? We can use the `ShardRowIDBits` to record the max shard row ID bit. Then we can remove the field of `MaxShardRowIDBits`.